### PR TITLE
Cleanup pthread JS->native symbol dependencies. NFC

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

**Review by kleisauke — APPROVED** [review]
LGTM! CI failures can be resolved by rebaselining `other.test_metadce_minimal_pthreads`, I think.

File: emcc.py [hunk 1]
```diff
@@ -1933,39 +1933,37 @@ def default_setting(name, new_default):
...
-    building.user_requested_exports.add('__emscripten_thread_exit')
-    building.user_requested_exports.add('_emscripten_tls_init')
-    building.user_requested_exports.add('_emscripten_current_thread_process_queued_calls')
+    worker_imports = [
+      '__emscripten_thread_init',
+      '__emscripten_thread_exit',
+      '_emscripten_tls_init',
+      '_emscripten_futex_wake',
+      '_emscripten_current_thread_process_queued_calls',
+      '_pthread_self',
```
Comment (kripken) [inline emcc.py:1963]: are `emscripten_futex_wait` and `pthread_self` really only used by `worker.js`?

I don't see any references from JS to the former, and I do see references from other files to the latter.

Comment (sbc100) [reply emcc.py:1963]: This list is for symbols that are needed by `worker.js`.    The existence a symbols in this list does not mean that the symbol is not also needed by JS.    These symbols are added to both `EXPORTED_FUNCTIONS` (which keeps them alive from the JS POV,  and `user_requested_exports` which protexts from the metadce.

Comment (kripken) [reply emcc.py:1963]: I guess that makes sense. It slightly worries me though that I read this code as "the worker.js section", which could lead to me thinking I can remove something there if it's not in worker.js.

**Pull request merged**